src/twisted/mail/newsfragments/12345.bugfix

```diff
@@ -0,0 +1 @@
+Fix over length partial response for iOS mail.app
```

Comment:
I forgot to comment this in the initial review.

For the release notes, we prefer to mention the affected public API.

I think that this is the affected public API.

I think that this is a better explanation of the change

The newsfragments are designed to be used as release notes... and not as commit message.

```suggestion
twisted.mail.imap.IMAPServer.arg_fetchatt can now detect that a IMAP partial request has been made that is longer than the original message and alter the output of the response in order to match the IMAPRev4 spec.
```